packages/wonder-blocks-dropdown/components/dropdown-opener.js

```diff
@@ -33,32 +35,46 @@ type Props = {|
      * Test ID used for e2e testing.
      */
     testId?: string,
+
+    /**
+     * Text for the opener that can be passed to the child as an argument.
+     */
+    text: string,
 |};
 
 class DropdownOpener extends React.Component<Props> {
     static defaultProps = {
         disabled: false,
     };
 
+    getTestIdFromProps = (childrenProps: any) => {
+        return childrenProps.testId || childrenProps["data-test-id"];
+    };
+
     renderAnchorChildren(
         eventState: ClickableState,
         handlers: ClickableHandlers,
     ) {
-        const renderedChildren = this.props.children(eventState);
+        const {disabled, testId, text} = this.props;
+        const renderedChildren = this.props.children(eventState, text);
+        const childrenProps = renderedChildren.props;
+        const childrenTestId = this.getTestIdFromProps(childrenProps);
 
         return React.cloneElement(renderedChildren, {
             ...handlers,
-            disabled: this.props.disabled,
-            "data-test-id": this.props.testId,
-            onClick: renderedChildren.props.onClick
+            disabled,
+            onClick: childrenProps.onClick
                 ? (e: SyntheticMouseEvent<>) => {
                       // This is done to avoid overriding a
                       // custom onClick handler inside the
                       // children node
-                      renderedChildren.props.onClick(e);
+                      childrenProps.onClick(e);
                       handlers.onClick(e);
                   }
                 : handlers.onClick,
+            // try to get the testId from the child element
+            // If it's not set, try to fallback to the parent's testId
+            "data-test-id": childrenTestId || testId,
```

Comment:
@kevinbarabash I've added this extra check to get the `testId` or the `data-test-id` value from the opener element. There's an special case (`ActionMenu, default opener`) where we need to pass the `testId` from the parent element. In the other cases, if we don't set the ID from the custom one, we won't be passing any `testId` to the cloned element at all.